R/helpers_data.R

```diff
@@ -0,0 +1,235 @@
+#'
+#' A method that creates a plot for [`Data`]  and [`DataOrdinal`] objects.
+#'
+#' @param x (`Data`)\cr object we want to plot.
+#' @param y (`missing`)\cr missing object, for compatibility with the generic
+#'   function.
+#' @param blind (`flag`)\cr indicates whether to blind the data.
+#'   If `TRUE`, then placebo subjects are reported at the same level
+#'   as the active dose level in the corresponding cohort,
+#'   and DLTs are always assigned to the first subjects in a cohort.
+#' @param tox_labels (`character`)\cr the toxicity category labels
+#' @param tox_shapes (`integer`)\cr the shapes used to plot the various toxicity
+#' categories
+#' @param legend (`flag`)\cr whether the legend should be added.
+#' @param ... not used.
+#'
+#' @note The default values of `tox_shapes` and `tox_labels` result in DLTs
+#' being displayed as red triangles and other responses as black circles.
+#' @return The [`ggplot2`] object.
+#'
+#' @rdname plot-Data
+h_plot_data_dataordinal <- function(
+    x,
+    blind = FALSE,
+    legend = TRUE,
+    tox_labels = c(Yes = "red", No = "black"),
+    tox_shapes = c(Yes = 17L, No = 16L),
+    ...) {
+  assert_flag(blind)
+  assert_flag(legend)
+  assert_character(tox_labels, any.missing = FALSE, unique = TRUE)
+  assert_integer(tox_shapes, any.missing = FALSE, unique = TRUE)
+  assert_true(length(tox_shapes) == length(tox_labels))
+  assert_subset(x@y, as.integer(0:(length(tox_shapes) - 1)))
+  if (x@nObs == 0L) {
+    return()
+  }
+  df <- h_plot_data_df(x, blind, ...)
+
+  p <- ggplot(df, aes(x = patient, y = dose)) +
+    geom_point(aes(shape = toxicity, colour = toxicity), size = 3) +
+    scale_colour_manual(
+      name = "Toxicity",
+      values = tox_labels,
+      breaks = names(tox_labels),
+      guide = guide_legend(reverse = TRUE)
+    ) +
+    scale_shape_manual(
+      name = "Toxicity",
+      values = tox_shapes,
+      breaks = names(tox_shapes),
+      guide = guide_legend(reverse = TRUE)
+    ) +
+    scale_x_continuous(breaks = df$patient, minor_breaks = NULL) +
+    scale_y_continuous(
+      breaks = sort(unique(c(0, df$dose))),
+      minor_breaks = NULL,
+      limits = c(0, max(df$dose) * 1.1)
+    ) +
+    xlab("Patient") +
+    ylab("Dose Level")
+
+  p <- p + h_plot_data_cohort_lines(df$cohort, placebo = x@placebo)
+
+  if (!blind) {
+    p <- p +
+      geom_text(
+        aes(label = ID, size = 2),
+        data = df,
+        hjust = 0,
+        vjust = 0.5,
+        angle = 90,
+        colour = "black",
+        show.legend = FALSE
+      )
+  }
+
+  if (!legend) {
+    p <- p + theme(legend.position = "none")
+  }
+
+  p
+}
+
+#' Helper Function Containing Common Functionality
+#'
+#' Used by `dose_grid_range-Data` and `dose_grid_range-DataOrdinal`
+#' @rdname h_obtain_dose_grid_range
+#'
+#' @param object (`Data` or `DataOrdinal`)\cr the object for which the dose grid
+#' range is required
+#' @param ignore_placebo (`flag`)\cr should placebo dose (if any) not be counted?
```

Comment:
also with `@rdname` we should not overwrite previous argument descriptions

Reply:
`#' @rdname h_obtain_dose_grid_range` is unique.  There are no "previous argument descriptions" to overwrite.  Does this make `@rdname` redundant?

Reply:
Yes, if this documentation page is completely stand alone, then we should not use the `@rdname` tag.
(Note that there could be an existing documentation page with this name that does not explicitly use `@rdname`, so good to double check)

Reply:
Done.